### PR TITLE
Fixed init script for Amazon linux 

### LINKED ISSUE #214: haproxy service not properly configured through AWS OpsWorks stacks
### Cookbook version
4.1.0

### Chef-client version
chef-client 12.18.31
OpsWorks Stacks uses Chef Zero (logs show 12.13.37)

### Platform Details
The configuration is done using the latest Amazon AMI with OpsWorks Stacks

### Scenario:
I try to achieve a very basic installation of haproxy from sources using AWS OpsWorks Stacks.
Note that everything is fine when executed the recipe with Kitchen on centos 6.8 (I tried this because Amazon AMI seems to be derivated from centos 6).

### Steps to Reproduce:
Create an OpsWorks Stacks layer configured to run a very basic recipe with all default values performing the installation from sources, and launch an ec2 instance with the latest AMI.
There may be another simpler way.

### Expected Result:
The haproxy service should be up and running

### Actual Result:
The service cannot start because the exec path is invalid. 
The resulting init.d/haproxy file shows
`
config="/haproxy.cfg"
exec="/sbin/haproxy"
`

The variables @conf_dir and @prefix seems to be left blank

### PR DESCRIPTION
**Pull request by rshade**
### Description
updated the config script for Amazon linux
### Issues Resolved
#214 

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by rshade** [comment]
[root@haproxy-amazon ~]# ps auxww | grep haproxy
haproxy   3360  0.0  0.0  50396   932 ?        Ss   17:02   0:00 /usr/sbin/haproxy -D -f /etc/haproxy/haproxy.cfg -p /var/run/haproxy.pid
root      4024  0.0  0.0 110460  2096 pts/0    S+   17:05   0:00 grep --color=auto haproxy
[root@haproxy-amazon ~]#

File: .kitchen.dokken.yml [hunk 1]
```diff
@@ -67,6 +67,15 @@ platforms:
...
       - RUN zypper --non-interactive install aaa_base perl-Getopt-Long-Descriptive which hostname
 
+- name: amazon
+  driver:
+    image: amazonlinux:latest
+    pid_one_command: /sbin/init
+    intermediate_instructions:
+      - RUN yum update -y
+      - RUN yum -y install which initscripts net-tools wget upstart procps util-linux
+      - RUN echo -e "NETWORKING=yes\nNETWORKING_IPV6=no\nHOSTNAME=localhost.localdomain\n" > /etc/sysconfig/network
```
Comment (shortdudey123) [inline .kitchen.dokken.yml]: any reason to disable ipv6 in testing? is it disabled by default in the AMI's?

**Comment by rshade** [comment]
[2017-05-03T21:19:16+00:00] INFO: Processing service[haproxy] action restart (/var/chef/runs/8a61731d-935a-4777-8106-c39fb4c5f030/local-mode-cache/cache/cookbooks/haproxy/resources/install.rb line 130)
[2017-05-03T21:19:16+00:00] INFO: service[haproxy] restarted
[2017-05-03T21:19:16+00:00] INFO: Chef Run complete in 13.960746653 seconds
[2017-05-03T21:19:16+00:00] INFO: Running report handlers
[2017-05-03T21:19:16+00:00] INFO: Report handlers complete

**Comment by rshade** [comment]
tested with: https://github.com/rshade/opsworks-haproxy/blob/master/recipes/package.rb, https://github.com/rshade/opsworks-haproxy/blob/master/recipes/source.rb

**Pull request merged**